devices/Bmm150/Bmm150.cs

```diff
@@ -0,0 +1,359 @@
+        /// </summary>
+        /// <param name="numberOfMeasurements">Number of measurement for the calibration, default is 100</param>
+        // https://platformio.org/lib/show/12697/M5_BMM150
+        public void CalibrateMagnetometer(int numberOfMeasurements = 100)
+        {
+            Vector3 mag_min = new Vector3() { X = 9000, Y = 9000, Z = 30000 };
+            Vector3 mag_max = new Vector3() { X = -9000, Y = -9000, Z = -30000 };
+            Vector3 rawMagnetometerData;
+
+            for (int i = 0; i < numberOfMeasurements; i++)
+            {
+                try
+                {
+                    rawMagnetometerData = ReadMagnetometerWithoutCorrection();
+
+                    if (rawMagnetometerData.X != 0)
+                    {
+                        mag_min.X = (rawMagnetometerData.X < mag_min.X) ? rawMagnetometerData.X : mag_min.X;
+                        mag_max.X = (rawMagnetometerData.X > mag_max.X) ? rawMagnetometerData.X : mag_max.X;
+                    }
+
+                    if (rawMagnetometerData.Y != 0)
+                    {
+                        mag_max.Y = (rawMagnetometerData.Y > mag_max.Y) ? rawMagnetometerData.Y : mag_max.Y;
+                        mag_min.Y = (rawMagnetometerData.Y < mag_min.Y) ? rawMagnetometerData.Y : mag_min.Y;
+                    }
+
+                    if (rawMagnetometerData.Z != 0)
+                    {
+                        mag_min.Z = (rawMagnetometerData.Z < mag_min.Z) ? rawMagnetometerData.Z : mag_min.Z;
+                        mag_max.Z = (rawMagnetometerData.Z > mag_max.Z) ? rawMagnetometerData.Z : mag_max.Z;
+                    }
+
+                    Wait(100);
+                }
+                catch
+                {
+                    // skip this reading
+                }
+            }
+
+            _calibrationCompensation.X = (mag_max.X + mag_min.X) / 2;
+            _calibrationCompensation.Y = (mag_max.Y + mag_min.Y) / 2;
+            _calibrationCompensation.Z = (mag_max.Z + mag_min.Z) / 2;
+        }
+
+        /// <summary>
+        /// True if there is a data to read
+        /// </summary>
+        public bool HasDataToRead => (ReadByte(Register.DATA_READY_STATUS) & 0x01) == 0x01;
+
+        /// <summary>
+        /// Check if the version is the correct one (0x48). This is fixed for this device
+        /// Page 28 from the documentation :
+        /// Device ID of AKM. It is described in one byte and fixed value.  48H: fixed
+        /// </summary>
+        /// <returns>Returns true if the version match</returns>
+        public bool IsVersionCorrect()
+        {
+            return ReadByte(Register.WIA) == 0x32;
+        }
+
+        /// <summary>
+        /// Read the magnetometer without Bias correction and can wait for new data to be present
+        /// </summary>
+        /// <remarks>
+        /// Vector axes are the following:
+        ///         +X
```

Comment:
Are you sure about the axes representation? Please double and triple check it is the correct one

Reply:
Good catch, this was copied from the AK89xx sensor. I will remove it from here.